AMP-8 login page

AniketRaj314 opened the pull request:
TODO:
- Send Request to API for verifying login
- Add Form Validation
- Improve UI

AniketRaj314 committed:
Added Bootstrap, Reactstrap and Framer as packages
Included base login component to App.js

AniketRaj314 committed:
Login base form setup complete

Need to add validation and send request to API

ksdfg commented:
@vercel[bot] eeeek why failed

akhilnarang commented:
![image](https://user-images.githubusercontent.com/10119037/102003980-429a4e00-3d32-11eb-82b0-db01a01666ec.png)

AniketRaj314 commented:
Ahh, I see. Lemme update that fir. Thanks. Where did you find the logs though?

akhilnarang commented:
My vercel account, so it shows for me :3
Will see if I can create a combined one.

AniketRaj314 committed:
Removed unnecessary imports

Label, Button and logo were not necessary

ksdfg commented:
![image](https://user-images.githubusercontent.com/43273771/102008463-af274400-3d56-11eb-9484-9364bf9d7fe2.png)

incomplete page, or rekt CSS?

AniketRaj314 committed:
Deleted package-lock.json to create conflicts with yarn.lock

AniketRaj314 committed:
Send POST call with login credentials to get a token

TODO (this branch):
- Error handling for 400 and 422
- Fix UI

AniketRaj314 committed:
Improved UI using CSS Flexbox

Works way better than bootstrap.

TODO:
- Set a getUser method to verify Auth Token
- Display Appropriate Error Messages

AniketRaj314 committed:
Created a getUser to get the user object

Logging in console for now, will use react-toastify for toasts (next
commit)

Reduced transition duration and made the submit button a bit more
prominent

AniketRaj314 committed:
Implemented toasts for successful login and errors

Using react-toastify to display toasts and will be implemented
throughout the platform instead of modals (hopefully)

TODO:
- Display error messages accurately once statusText has been updated or
  we find a sane way to access the detail field in the response

- Add Email and Password Validation

- (not important rn) Add a loader while authenticating login

akhilnarang committed:
login: a few small changes

Try to fix display of the error message

References taken from https://stackoverflow.com/questions/29473426/fetch-reject-promise-with-json-error-object

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

AniketRaj314 commented on src/pages/login/login.page.jsx:
```diff
@@ -35,14 +35,12 @@ class Login extends Component {
             })
             .then(token_object => this.setState({ user_auth_token: token_object.access_token }, this.getUser))
-            .catch(error => toast.error(`Error 😓: ${error.message}`, {
+            .catch(error => toast.error(`Error 😓: ${error['detail']}`, {
```
LGTM, just can we make it error.detail for consistency purpose?

AniketRaj314 reviewed (commented):
RIP I wrote same code, but forgot to put async before response in the params.

akhilnarang committed:
login: `error['detail']` -> `error.detail`

As suggested by Aniket

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

AniketRaj314 committed:
React App => REVAMP

Will keep changing titles as the pages change

AniketRaj314 committed:
Validation for Email ID and Password added

Branch ready to merge

TODO:
- Add a loader

AniketRaj314 committed:
Cleaned up console logs and replicated error handling for getUser
